utils.go

```diff
@@ -155,6 +157,22 @@ func (h *Headscale) getAvailableIPs() (MachineAddresses, error) {
 	return ips, err
 }
 
+func (h *Headscale) getRandomAvailableIPs() (MachineAddresses, error) {
+	var ips MachineAddresses
+	var err error
+	ipPrefixes := h.cfg.IPPrefixes
+	for _, ipPrefix := range ipPrefixes {
+		var ip *netip.Addr
+		ip, err = h.getAvailableIP(ipPrefix)
+		if err != nil {
+			return shuffleIPs(ips), err
+		}
+		ips = append(ips, *ip)
+	}
+
+	return shuffleIPs(ips), err
```

Comment:
The error is being handled in the loop, it will always be nil and it should be explicit as such.

```go
return shuffleIPs(ips), nil
```